### PR TITLE
NIAD-1395 linkset observation statement code populated

### PR DESCRIPTION
**Pull request by dawidskainos**

### CODE REVIEW

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ConditionLinkSetMapper.java [hunk 1]
```diff
@@ -203,4 +207,11 @@ private boolean filterOutListResourceType(Reference reference) {
     private boolean checkIfReferenceIsObservation(Reference reference) {
         return reference.getReferenceElement().getResourceType().equals(ResourceType.Observation.name());
     }
+
+    private String buildCode(Condition condition) {
+        if (condition.hasCode()) {
+            return codeableConceptCdMapper.mapCodeableConceptToCd(condition.getCode());
+        }
+        throw new EhrMapperException("Condition code not present");
```
Comment (andrewwa-kainos) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ConditionLinkSetMapper.java:215]: Can we get a test for the exception scenario too?

**Review by andrewwa-kainos — CHANGES_REQUESTED** [review]

**Review by n-r-b — APPROVED** [review]
Looks good - just need the test covering the exception scenerio

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ConditionLinkSetMapper.java (hunk 1 above)
Comment (dawidskainos) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ConditionLinkSetMapper.java:215]: done

**Review by z-uddin — DISMISSED** [review]

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ConditionLinkSetMapperTest.java [hunk 2]
```diff
@@ -233,4 +241,14 @@ public void When_MappingParsedConditionWithNonExistentReferenceInRelatedClinical
             .isExactlyInstanceOf(EhrMapperException.class)
             .hasMessage("Could not resolve Condition Related Medical Content reference");
     }
+
+    @Test
+    public void When_MappingParsedConditionCodeIsMissing_Expect_MapperException() throws IOException {
+        var jsonInput = ResourceTestFileUtils.getFileContent(INPUT_JSON_MISSING_CONDITION_CODE);
+        Condition parsedObservation = fhirParseService.parseResource(jsonInput, Condition.class);
+
+        assumeThatThrownBy(() -> conditionLinkSetMapper.mapConditionToLinkSet(parsedObservation, false))
```
Comment (n-r-b) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ConditionLinkSetMapperTest.java:250]: There's no need to assume here, we can use assertions instead
```suggestion
        assertThatThrownBy(() -> conditionLinkSetMapper.mapConditionToLinkSet(parsedObservation, false))
```

**Review by n-r-b — APPROVED** [review]

**Review by andrewwa-kainos — APPROVED** [review]

**Pull request merged**